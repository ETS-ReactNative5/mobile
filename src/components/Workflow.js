import React, { Component } from 'react';
import {
  TouchableOpacity,
  StyleSheet,
  View
} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet'
import StyledText from './StyledText'
import Icon from 'react-native-vector-icons/FontAwesome'
import PropTypes from 'prop-types';

class Workflow extends Component {
  render() {
      return (
      <TouchableOpacity
        style={styles.workflowRow}
        onPress={() => this.props.openMobileProject(this.props.workflow.id)}>
        <StyledText text={this.props.workflow.display_name} />
        <View style={[styles.workflowIconContainer, { backgroundColor: this.props.color }]}>
          <Icon name='angle-right' style={styles.workflowIcon} />
        </View>
      </TouchableOpacity>
    );
  }
}

const styles = EStyleSheet.create({
  $iconSize: 30,
  workflowRow: {
    borderTopColor: '$greyBorder',
    borderTopWidth: StyleSheet.hairlineWidth,
    paddingHorizontal: 15,
    paddingVertical: 5,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  workflowIconContainer: {
    borderRadius: '0.5 * $iconSize',
    alignItems: 'center',
    justifyContent: 'flex-start',
    width: '$iconSize',
    height: '$iconSize'
  },
  workflowIcon: {
    backgroundColor: 'transparent',
    color: 'white',
    fontSize: 24,
    lineHeight: 26,
    marginLeft: 2
  },
})

Workflow.propTypes = {
  workflow: PropTypes.shape({
    display_name: PropTypes.string,
    id: PropTypes.string
  }),
  color: PropTypes.string,
  openMobileProject: PropTypes.func
}

export default Workflow
