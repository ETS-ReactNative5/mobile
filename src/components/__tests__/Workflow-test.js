import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Workflow from '../Workflow'

const workflow = {
  id: '1',
  display_name: 'Nice workflow',
}

it('renders correctly', () => {
  const tree = renderer.create(
    <Workflow workflow={workflow} color={'#AFA48C'} openMobileProject={jest.fn}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
