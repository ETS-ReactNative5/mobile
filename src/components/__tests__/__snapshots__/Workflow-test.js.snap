exports[`test renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        undefined,
      ]
    }>
    Nice workflow
  </Text>
  <View
    style={
      Array [
        undefined,
        Object {
          "backgroundColor": "#AFA48C",
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "FontAwesome",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          undefined,
        ]
      }>
      ï„…
    </Text>
  </View>
</TouchableOpacity>
`;
